service/share/share.go

```diff
@@ -161,37 +161,37 @@ func (s *service) GetSharesByNamespace(ctx context.Context, root *Root, nID name
 		return nil, ipld.ErrNotFoundInRange
 	}
 
-	type res struct {
-		nodes []format.Node
-		err   error
-	}
-	resultCh := make(chan *res)
-
 	ctx, cancel := context.WithCancel(ctx)
 	defer cancel()
 
-	for _, rootCID := range rowRootCIDs {
-		go func(rootCID cid.Cid) {
-			nodes, err := ipld.GetLeavesByNamespace(ctx, s.dag, rootCID, nID)
-			resultCh <- &res{nodes: nodes, err: err}
-		}(rootCID)
+	errCh := make(chan error)
+	nodes := make([][]format.Node, len(rowRootCIDs))
+	for i, rootCID := range rowRootCIDs {
+		go func(i int, rootCID cid.Cid) {
+			var err error
+			nodes[i], err = ipld.GetLeavesByNamespace(ctx, s.dag, rootCID, nID)
+			errCh <- err
+		}(i, rootCID)
 	}
 
-	namespacedShares := make([]Share, 0)
 	for i := 0; i < len(rowRootCIDs); i++ {
 		select {
-		case result := <-resultCh:
-			if result.err != nil {
-				return nil, result.err
-			}
-			for _, node := range result.nodes {
-				namespacedShares = append(namespacedShares, node.RawData()[1:])
+		case err := <-errCh:
+			if err != nil {
+				return nil, err
```

Comment:
nit: In this case now, it would be nicer/cleaner to use [errgoup](https://pkg.go.dev/golang.org/x/sync/errgroup) instead, but not necessary.

Reply:
Oh, I didn't know you use errgroup in node :) Will fix.